crates/cubecl-core/src/pod.rs

```diff
@@ -45,11 +46,8 @@ impl CubeElement for u32 {
     fn type_name() -> &'static str {
         "u32"
     }
-    fn as_bytes(slice: &[Self]) -> &[u8] {
-        bytemuck::cast_slice(slice)
-    }
-    fn from_bytes(bytes: &[u8]) -> &[Self] {
-        bytemuck::cast_slice(bytes)
+    fn from_elem_data(bytes: Vec<u8>) -> Vec<Self> {
+        bytemuck::cast_slice(bytes.as_slice()).to_vec()
```

Comment:
That can be very very slow, I used unsafe in `TensorData` in Burn to avoid using this.